pkg/gofr/http/responder.go

```diff
@@ -49,40 +69,67 @@ func (r Responder) Respond(data interface{}, err error) {
 }
 
 // getStatusCode returns corresponding HTTP status codes.
-func getStatusCode(method string, data interface{}, err error) (status int, errObj interface{}) {
+func getStatusCode(method string, data interface{}, err error) (status int) {
 	if err == nil {
 		switch method {
 		case http.MethodPost:
 			if data != nil {
-				return http.StatusCreated, nil
+				return http.StatusCreated
 			}
 
-			return http.StatusAccepted, nil
+			return http.StatusAccepted
 		case http.MethodDelete:
-			return http.StatusNoContent, nil
+			return http.StatusNoContent
 		default:
-			return http.StatusOK, nil
+			return http.StatusOK
 		}
 	}
 
-	e, ok := err.(statusCodeResponder)
-	if ok {
-		return e.StatusCode(), map[string]interface{}{
-			"message": err.Error(),
+	var e statusCodeResponder
+	if errors.As(err, &e) {
+		if data != nil {
+			return http.StatusPartialContent
+		}
+
+		status = e.StatusCode()
+
+		if e.StatusCode() == 0 {
+			return http.StatusInternalServerError
 		}
-	}
 
-	return http.StatusInternalServerError, map[string]interface{}{
-		"message": err.Error(),
+		return status
 	}
-}
 
-// response represents an HTTP response.
-type response struct {
-	Error interface{} `json:"error,omitempty"`
-	Data  interface{} `json:"data,omitempty"`
+	return http.StatusInternalServerError
 }
 
-type statusCodeResponder interface {
-	StatusCode() int
+func getErrResponse(err error) []errResponse {
+	var (
+		errResp []errResponse
+		m       MultipleErrors
+		c       CustomError
+	)
+
+	if err != nil {
```

Comment:
```suggestion
	if err == nil {
	    return nil
	}
```

Reply:
please go through the logic

Reply:
I dislike function wide if branch, but OK. Everyone has his style